src/Shared.TextureContainer/Shared.ImageHelper.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace KK_Plugins
+{
+    //Shamelessly stolen from https://stackoverflow.com/a/1246008
+    public static class ImageHelper
+    {
+        /// <summary>
+        /// Read the file signature from a byte array to deterimine its file format.
+        /// https://www.garykessler.net/library/file_sigs.html
+        /// </summary>
+        /// <param name="imageBytes">Byte array containing the image</param>
+        /// <returns>ImageFormat</returns>
+        public static ImageFormat GetContentType(byte[] imageBytes)
+        {
+            MemoryStream ms = new MemoryStream(imageBytes);
+
+            using (BinaryReader br = new BinaryReader(ms))
+            {
+                int maxMagicBytesLength = imageFormatDecoders.Keys.OrderByDescending(x => x.Length).First().Length;
+
+                byte[] magicBytes = new byte[maxMagicBytesLength];
+
+                for (int i = 0; i < maxMagicBytesLength; i += 1)
+                {
+                    magicBytes[i] = br.ReadByte();
+
+                    foreach (var kvPair in imageFormatDecoders)
+                    {
+                        if (magicBytes.StartsWith(kvPair.Key))
```

Comment:
Forgive me for chiming in
It's not related to WebP, and right now it's harmless... I was curious.
The number of bytes read is not taken into account. Wouldn't all data with the first 1byte of 0x00 be Avif?

How about something like this?
```
public static ImageFormat GetContentType(byte[] imageBytes)
{
    foreach( var kvPair in imageFormatDecoders )
    {
        var magicBytes = kvPair.Key;
        if (magicBytes.Length <= imageBytes.Length && imageBytes.StartsWith(magicBytes))
            return kvPair.Value;
    }
    return ImageFormat.Unrecognized;
}
```

Reply:
Thanks! Now that I take a good look, that would indeed be the case. Your code also looks a lot simpler to digest. I think this is even better:
```csharp
public static ImageFormat GetContentType(byte[] imageBytes)
{
    foreach (var kvPair in imageFormatDecoders.OrderByDescending(x => x.Key.Length))
        if (kvPair.Key.Length <= imageBytes.Length && imageBytes.StartsWith(kvPair.Key))
            return kvPair.Value;
    return ImageFormat.Unrecognized;
}
```

Now it handles larger magic number sequences first, which should solve most issues (if such clashing file formats were to ever be supported)

Reply:
I see. Indeed.
If the magics are compatible, I'll have to start looking at the long ones.

Reply:
resolved